components/script/dom/globalscope.rs

```diff
@@ -3065,7 +3065,19 @@ impl GlobalScope {
                 // of imageBitmap's bitmap to false.
                 image_bitmap.set_origin_clean(image.same_origin(GlobalScope::entry().origin()));
 
-                p.resolve_native(&image_bitmap, can_gc);
+                let trusted_promise = TrustedPromise::new(p.clone());
```

Comment:
Can we add a helper to encapsulate this, that takes a &ImageBitmap and a Rc<Promise> argument? It looks like it's duplicated a bunch of times.

Reply:
Yep, it looks better with helper closure...